internal/k3d/terraform.go

```diff
@@ -53,3 +59,62 @@ type GithubTerraformEnvs struct {
 	AwsAccessKeyId        string
 	AwsSecretAccessKey    string
 }
+
+// TerraformPrep prepares Terraform files by detokenizing them.
+// It processes files in the specified GitOps directory and the GitHub runner path.
+func TerraformPrep(config *K3dConfig) error {
```

Comment:
`internal/controller/terraform.go` and this seem to have the same functions, is there any way you see where we can define them once?

Reply:
if we need to make one function,we need to pass more parameters because the struct being used is not the same as other clouds and also want to keep k3d away from other clouds,but lets sync tomorrow to get a better approach :)